layers/dot11.go

```diff
@@ -2061,6 +2086,11 @@ func decodeDot11MgmtAction(data []byte, p gopacket.PacketBuilder) error {
 
 func (m *Dot11MgmtAction) LayerType() gopacket.LayerType  { return LayerTypeDot11MgmtAction }
 func (m *Dot11MgmtAction) CanDecode() gopacket.LayerClass { return LayerTypeDot11MgmtAction }
+func (m *Dot11MgmtAction) NextLayerType() gopacket.LayerType {return LayerTypeDot11InformationElement}
+func (m *Dot11MgmtAction) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+	m.Payload = data
```

Comment:
This is is why `TestPacketDot11MgmtAction` fails.
The frame in this test has 2 fixed parameters (category code and action) and after that one `LayerTypeDot11InformationElement` followed by 4 byte `FCS`.
In your version the two fixed parameters get decoded as `LayerTypeDot11InformationElement`, which is wrong and thus the incorrect length.
Sadly I can't find a good source for how this is done correctly (https://raw.githubusercontent.com/wireshark/wireshark/a5faef2e4fe9c8bb4b55dcb3e60c32a2a504ac9f/epan/dissectors/packet-ieee80211.c is way too much to read for reviewing this)